resources/function_help/json/to_int

```diff
@@ -2,14 +2,20 @@
   "name": "to_int",
   "type": "function",
   "groups": ["Conversions"],
-  "description": "Converts a string to integer number. Nothing is returned if a value cannot be converted to integer (e.g '123asd' is invalid).",
+  "description": "Rounds a numeric value to integer or converts a string to integer number. Nothing is returned if the value cannot be converted to integer (e.g '123asd' or '123.45' is invalid).",
   "arguments": [{
```

Comment:
Why is 123.45 invalid?

Reply:
it is '123.45', as string. Why invalid? I can't explain, I just observed. maybe a left-over?

![image](https://user-images.githubusercontent.com/7983394/187089499-ccfd1263-67c7-432e-af79-5baf40e09fb2.png)

Reply:
Hmm, that's very confusing. The function can convert a string to number, OR a double to into BUT not both?

Reply:
It seems a Qt feature/issue (`to_int `uses `QgsExpressionUtils::getIntValue` which uses `QVariant.toLongLong`)...

```Python
val = QVariant(10)
val.convert(QMetaType.LongLong)
True
val.value()
10

val = QVariant('10')
val.convert(QMetaType.LongLong)
True
val.value()
10

val = QVariant(10.6)
val.convert(QMetaType.LongLong)
True
val.value()
11

val = QVariant('10.6')
val.convert(QMetaType.LongLong)
False
val.value()
NULL

```

Reply:
Would it be safe to check for decimal separator, split the string, int the first half and increase if the bottom is lower? Or do a string to float to int?

Reply:
> Or do a string to float to int?

That sounds right 👍

Reply:
That's again beyond my skills. I welcome a commit or a replacement PR. Sorry.

Reply:
TBH I'm not sure if the safety would be better in `QgsExpressionUtils::getIntValue` or only in `fcnToInt`, as the first is used in many expressions internally, but I'm not sure if the added safety is needed.